### PR TITLE
Rss reader

### PR DESCRIPTION
**Pull request by dorooleg**

### CODE REVIEW

**Comment by thevery** [comment]
hint usage is wrong

**Comment by dorooleg** [comment]
Why?

**Comment by thevery** [comment]
hint should be like `enter RSS url, i.e. lenta.ru/rss`, but not `lenta.ru/rss` because it is confusing

**Comment by dorooleg** [comment]
Ok, i am agree

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -15,6 +17,13 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".ListArticlesActivity" />
+
+        <provider
+            android:name=".RSSContentProvider"
+            android:authorities="com.csc.rssreader.rssloader"
+            android:enabled="true"
+            android:exported="false"></provider>
```
Comment (thevery) [inline app/src/main/AndroidManifest.xml:26]: use self-closing

File: app/src/main/java/com/csc/rssreader/GlobalContext.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.csc.rssreader;
+
+/**
+ * Created by Oleg on 21.03.2016.
+ */
+public class GlobalContext {
+    private GlobalContext() {
+    }
+
+    public final static String KEY_URL = "com.csc.rssreader.KEY_URL";
```
Comment (thevery) [inline app/src/main/java/com/csc/rssreader/GlobalContext.java:10]: common naming is `EXTRA_*`

File: app/src/main/java/com/csc/rssreader/ListArticlesActivity.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            url = getString(R.string.default_rss);
+        }
+        listViewArticles = (ListView) findViewById(R.id.list_articles);
+        new RSSLoadTask(this).execute();
+    }
+
+    private static class RSSLoadTask extends AsyncTask<Void, Void, Void> {
+
+        private WeakReference<ListArticlesActivity> listArticlesActivityRef;
+        ArrayList<String> dates = new ArrayList<>();
```
Comment (thevery) [inline app/src/main/java/com/csc/rssreader/ListArticlesActivity.java:46]: why default visibility?

Comment (dorooleg) [inline app/src/main/java/com/csc/rssreader/ListArticlesActivity.java:46]: Good question))

File: app/src/main/java/com/csc/rssreader/ListArticlesActivity.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        ArrayList<String> titles = new ArrayList<>();
+        ArrayList<String> descriptions = new ArrayList<>();
```
Comment (thevery) [inline app/src/main/java/com/csc/rssreader/ListArticlesActivity.java:48]: reformat

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (dorooleg) [inline app/src/main/AndroidManifest.xml:26]: It is not the first time with me. I  will correct it in the following tasks. Thanks.

File: app/src/main/java/com/csc/rssreader/ListArticlesActivity.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            try {
+                cursor = activity.getContentResolver().query(Uri.parse("content://com.csc.rssreader.rssloader"), null, activity.url, null, null);
+                if (cursor != null) {
+                    while (cursor.moveToNext()) {
+                        dates.add(cursor.getString(RSSContentProvider.COLUMN_DATE));
+                        titles.add(cursor.getString(RSSContentProvider.COLUMN_TITLE));
+                        descriptions.add(cursor.getString(RSSContentProvider.COLUMN_DESCRIPTION));
+                    }
+                }
+            } catch (Exception ex) {
```
Comment (thevery) [inline app/src/main/java/com/csc/rssreader/ListArticlesActivity.java:68]: you should not ignore exception

File: app/src/main/java/com/csc/rssreader/ListArticlesActivity.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                    }
+                    if (descriptions.get(i) != null) {
+                        articleDescriptions[i] += descriptions.get(i);
+                    }
+                }
+                ListArticlesArrayAdapter adapter = new ListArticlesArrayAdapter(activity, articleDates, articleTitles, articleDescriptions);
+                activity.listViewArticles.setAdapter(adapter);
+            }
+            else {
+                if (activity != null)
```
Comment (thevery) [inline app/src/main/java/com/csc/rssreader/ListArticlesActivity.java:103]: missing `{}`

File: app/src/main/java/com/csc/rssreader/ListArticlesArrayAdapter.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Created by Oleg on 21.03.2016.
+ */
+public class ListArticlesArrayAdapter extends ArrayAdapter<String> {
+
+    private final Context context;
+    private final String[] articleDescriptions;
+    private final String[] articleTitles;
+    private final String[] articleDates;
```
Comment (thevery) [inline app/src/main/java/com/csc/rssreader/ListArticlesArrayAdapter.java:18]: such OOP

File: app/src/main/java/com/csc/rssreader/ListArticlesActivity.java (hunk 4 above)
Comment (dorooleg) [inline app/src/main/java/com/csc/rssreader/ListArticlesActivity.java:48]: What does it mean?

File: app/src/main/java/nl/matshofman/saxrssreader/RssHandler.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+				if(qName.equals("content:encoded")) 
+					qName = "content";
+				String methodName = "set" + qName.substring(0, 1).toUpperCase() + qName.substring(1);
+				Method method = rssItem.getClass().getMethod(methodName, String.class);
+				method.invoke(rssItem, stringBuilder.toString());
+			} catch (SecurityException e) {
+			} catch (NoSuchMethodException e) {
+			} catch (IllegalArgumentException e) {
+			} catch (IllegalAccessException e) {
+			} catch (InvocationTargetException e) {
```
Comment (thevery) [inline app/src/main/java/nl/matshofman/saxrssreader/RssHandler.java:92]: 0_o

File: app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+            android:imeOptions="actionGo"
+            android:inputType="textUri"
+            android:hint="@string/default_rss"/>
+
+        <Button
+            android:id="@+id/btn_ok"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:layout_marginLeft="7dp"
```
Comment (thevery) [inline app/src/main/res/layout/activity_main.xml:29]: why 7, not 8?

File: app/src/main/java/nl/matshofman/saxrssreader/RssReader.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+
+public class RssReader {
+
+    public static RssFeed read(URL url) throws SAXException, IOException {
+
+        return read(url.openStream());
```
Comment (thevery) [inline app/src/main/java/nl/matshofman/saxrssreader/RssReader.java:36]: close?

**Comment by thevery** [comment]
10 + 1 - 1 (code issues) = 10

File: app/src/main/java/nl/matshofman/saxrssreader/RssReader.java (hunk 10 above)
Comment (dorooleg) [inline app/src/main/java/nl/matshofman/saxrssreader/RssReader.java:36]: It's not my library))

File: app/src/main/java/com/csc/rssreader/ListArticlesActivity.java (hunk 4 above)
Comment (thevery) [inline app/src/main/java/com/csc/rssreader/ListArticlesActivity.java:48]: https://www.jetbrains.com/idea/help/reformatting-source-code.html

**Pull request closed without merging**